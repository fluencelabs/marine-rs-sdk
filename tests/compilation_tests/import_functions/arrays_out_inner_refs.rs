#![allow(improper_ctypes)]

use marine_rs_sdk::marine;

pub fn main() {}

#[marine]
#[module_import("arrays_passing_effector")]
extern "C" {
    #[marine]
    pub fn func_1() -> &String;

    #[marine]
    pub fn func_2() -> &Vec<Vec<Vec<Vec<u8>>>>;

    #[marine]
    pub fn func_3() -> Vec<&Vec<Vec<Vec<u8>>>>;

    #[marine]
    pub fn func_4() -> Vec<Vec<&Vec<Vec<u8>>>>;

    #[marine]
    pub fn func_5() -> Vec<Vec<Vec<&Vec<u8>>>>;

    #[marine]
    pub fn func_6() -> Vec<Vec<Vec<Vec<&u8>>>>;
}
