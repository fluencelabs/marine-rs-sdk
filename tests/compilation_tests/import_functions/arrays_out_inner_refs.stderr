error: import function can't return a value with references
  --> $DIR/arrays_out_inner_refs.rs:10:5
   |
10 |     #[marine]
   |     ^
